### PR TITLE
[SDP][Minor] Removing an assert which does not seem always accurate

### PR DESCRIPTION
**Pull request by blefaudeux**
# Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes https://fb.workplace.com/groups/pytorchLightning/permalink/1432835657052968/
I'm not super clear on why this happens, but the general thinking is that this was a hard guard on not loosing any gradient, and actually the user may want to do to whatever (like, but may not be all of it: no_sync()fw/bw, then .eval()).
Keep the warning so that it's somewhat visible

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: fairscale/nn/data_parallel/fully_sharded_data_parallel.py [hunk 1]
```diff
@@ -1664,9 +1664,7 @@ def wrap_bn_only_policy(module: nn.Module, recurse: bool, unwrapped_params: int)
         if recurse:
             return not isinstance(module, tuple(default_auto_wrap_policy.FORCE_LEAF_MODULES))  # type: ignore
         else:
-            return is_bn and not isinstance(
-                module, tuple(default_auto_wrap_policy.EXCLUDE_WRAP_MODULES)
-            )  # type: ignore
+            return is_bn and not isinstance(module, tuple(default_auto_wrap_policy.EXCLUDE_WRAP_MODULES))  # type: ignore
```
Comment (blefaudeux) [inline fairscale/nn/data_parallel/fully_sharded_data_parallel.py:1667]: unrelated, but master linting was broken because of that

**Review by min-xu-ai — APPROVED** [review]
warning makes sense.

File: tests/nn/data_parallel/test_sharded_ddp_features.py [hunk 2]
```diff
@@ -288,6 +288,14 @@ def test_train_eval_change():
     dist.destroy_process_group()
 
 
+def test_train_eval_change():
+    world_size = 4
```
Comment (blefaudeux) [inline tests/nn/data_parallel/test_sharded_ddp_features.py:292]: making the test case truly distributed, but still not seeing the issue which was reported in a somewhat similar situation. My guess is that there are some changes on the lightning side which make it a little more than just .train()/.eval(), but crashing on users is probably not a good idea in that case. Worst case is a lost step() worth of gradients
